framework/src/csg/CSGSurfaceList.C

```diff
@@ -0,0 +1,74 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CSGSurfaceList.h"
+#include "CSGPlane.h"
+#include "CSGSphere.h"
+#include "CSGXCylinder.h"
+#include "CSGYCylinder.h"
+#include "CSGZCylinder.h"
+
+namespace CSG
+{
+
+CSGSurfaceList::CSGSurfaceList() {}
+
+void
+CSGSurfaceList::checkSurfaceName(const std::string & name) const
+{
+  if (_surfaces.find(name) != _surfaces.end())
+    mooseError("Surface with name " + name + " already exists in geometry.");
+}
+
+CSGSurface &
+CSGSurfaceList::getSurface(const std::string & name) const
+{
+  auto surf = _surfaces.find(name);
+  if (surf == _surfaces.end())
+    mooseError("No surface by name " + name + " exists in the geometry.");
+  else
+    return *(surf->second);
+}
+
+std::vector<std::reference_wrapper<const CSGSurface>>
+CSGSurfaceList::getAllSurfaces() const
+{
+  std::vector<std::reference_wrapper<const CSGSurface>> surfaces;
+  for (auto it = _surfaces.begin(); it != _surfaces.end(); ++it)
+    surfaces.push_back(*(it->second));
+  return surfaces;
+}
+
+CSGSurface &
+CSGSurfaceList::addSurface(std::unique_ptr<CSGSurface> & surf)
+{
+  auto name = surf->getName();
+  checkSurfaceName(name);
+  _surfaces.insert(std::make_pair(name, std::move(surf)));
+  return *_surfaces[name];
+}
+
+void
+CSGSurfaceList::renameSurface(const CSGSurface & surface, const std::string & name)
```

Comment:
This a weird thing. Is this actually a common used use case?

Reply:
Yes, it is common to rename objects for CSG definitions. I could imagine a use case where you would want to rename things before joining bases. It shouldn't hurt to have it available as an option. It would be more detrimental I think to not allow this.